mod-data-export/src/test/resources/domain/dataexport/features/mapping-profiles.feature

```diff
@@ -83,3 +83,51 @@ Feature: Test mapping profiles
     Then status 403
     And match response contains 'Deletion of default mapping profile is forbidden'
 
+  Scenario: should return UnprocessableEntity response when post mapping profile with invalid transformations - invalid tag
+    Given path 'data-export/mapping-profiles'
+    * print read('classpath:samples/mapping-profile/mp-transformation_invalid_tag.json')
+    And def invalidMappingProfile = read('classpath:samples/mapping-profile/mp-transformation_invalid_tag.json')
+    And request invalidMappingProfile
+    When method POST
+    Then status 422
+
+  Scenario: should return UnprocessableEntity response when post mapping profile with invalid transformations - invalid index
+    Given path 'data-export/mapping-profiles'
+    And def invalidMappingProfile = read('classpath:samples/mapping-profile/mp-transformation_invalid_index.json')
+    And request invalidMappingProfile
+    When method POST
+    Then status 422
+
+  Scenario: should return UnprocessableEntity response when post mapping profile with invalid transformations - invalid subfield
+    Given path 'data-export/mapping-profiles'
+    And def invalidMappingProfile = read('classpath:samples/mapping-profile/mp-transformation_invalid_subfield.json')
+    And request invalidMappingProfile
+    When method POST
+    Then status 422
+
+  Scenario: should return UnprocessableEntity response when post mapping profile with empty transformation and item as record type
+    Given path 'data-export/mapping-profiles'
+    And def invalidMappingProfile = read('classpath:samples/mapping-profile/mp-empty-transformation-item-record-type.json')
+    And request invalidMappingProfile
+    When method POST
+    Then status 422
+
+  Scenario: should return OK response when post mapping profile with empty transformation and holdings as transformation record type
+    Given path 'data-export/mapping-profiles'
+    And def validMappingProfile = read('classpath:samples/mapping-profile/mp-empty-transformation-holding-record-type.json')
+    And set validMappingProfile.id = uuid()
+    And set validMappingProfile.name = randomString(10)
+    * print validMappingProfile
+    And request validMappingProfile
+    When method POST
+    Then status 201

```

Comment:
Should we then `GET` the entity after and verify it's state?

Reply:
The POST method here returns the created entity as JSON and we can verify it here without additional GET call, I added some cheks. As well, the suggested testing behavior with follow-up GET calls already present above in a file. I just added specific cases.